pantalaimon/daemon.py

```diff
@@ -1039,19 +1079,44 @@ async def search(self, request):
 
         return web.json_response(result, headers=CORS_HEADERS, status=200)
 
-    async def download(self, request):
-        server_name = request.match_info["server_name"]
-        media_id = request.match_info["media_id"]
-        file_name = request.match_info.get("file_name")
+    async def upload(self, request):
+        file_name = request.query.get("filename", "")
+        content_type = request.headers.get("Content-Type", "application/octet-stream")
+        client = next(iter(self.pan_clients.values()))
+
+        try:
+            response = await client.upload(
+                data_provider=await request.read,
+                content_type=content_type,
+                filename=file_name,
+                encrypt=True,
+            )
+
+            if not isinstance(response, UploadResponse):
```

Comment:
This will never be true, no? The return value of `client.upload()` is always a tuple.